### PR TITLE
Deploy to both docker accounts

### PR DESCRIPTION
**Pull request by MuriloDalRi**
We are moving to the governmentdigitalservice account but we want to make sure everything has been migrated before switching off the govuk account.

https://trello.com/c/6JRYK8hU/2982-push-to-pull-from-governmentdigitalservice-docker-org-3

### CODE REVIEW

File: vars/govuk.groovy [hunk 1]
```diff
@@ -960,7 +963,15 @@ def dockerTagMasterBranch(jobName, branchName, buildNumber) {
 def pushDockerImage(imageName, tagName, asTag = null) {
   tagName = safeDockerTag(tagName)
   docker.withRegistry('https://index.docker.io/v1/', 'govukci-docker-enterprise-hub') {
+    // Push to both accounts until we migrate everything to the new one
     docker.image("govuk/${imageName}:${tagName}").push(asTag ?: tagName)
+    try {
+      docker.image("governmentdigitalservice/${imageName}:${tagName}").ls
```
Comment (ChrisBAshton) [inline vars/govuk.groovy:969]: We don't seem to be doing anything with this `.ls` call - if we attempt to push on line 970 and it errors, we catch the error on line 971, so it doesn't look like we're doing any sort of check here.

Probably better to aim for this sort of logic if you can:

```groovy
  if (docker.image("governmentdigitalservice/${imageName}:${tagName}").ls) {
    docker.image("governmentdigitalservice/${imageName}:${tagName}").push(asTag ?: tagName)
  }
  else {
    println("governmentdigitalservice/${imageName} doesn't exist. If you want to push to this org, you must create the repo manually first.")
  }
```

**Review by ChrisBAshton — COMMENTED** [review]
Looking good 👍 
Just to clarify, we're not intending to push repos to `governmentdigitalservice` if they don't already exist (as we've only transferred the ones that we want to keep; anything else is just noise).
Reading some of the comments on Trello, it could be interpreted that you're intending to migrate _all_ govuk repos to governmentdigitalservice before switching off the govuk version. Just checking that's not the plan!

File: vars/govuk.groovy (hunk 1 above)
Comment (MuriloDalRi) [reply vars/govuk.groovy:969]: I tried doing that but the problem is that it [fails miserably](https://ci.integration.publishing.service.gov.uk/job/smartanswers/job/docker-test/9/console). The ls call is just so the logs look cleaner but not really necessary, so I can remove it.

Comment (MuriloDalRi) [reply vars/govuk.groovy]: Does it look better now? @ChrisBAshton

Comment (ChrisBAshton) [reply vars/govuk.groovy]: LGTM 👍 , thanks, Murilo

Could you also please answer the question from https://github.com/alphagov/govuk-jenkinslib/pull/113#pullrequestreview-1132808543? Then I'll be happy to approve. Thanks!

Comment (MuriloDalRi) [reply vars/govuk.groovy]: I was under the impression that we wanted to create the repos if they don't exist, but I think that has to be done manually anyway so this code should only push to existing repos or show the error message if the repo doesn't exist.

Comment (ChrisBAshton) [reply vars/govuk.groovy]: We only want to push the 23 repos identified in [the audit](https://trello.com/c/D7kUe72w/2961-investigate-govuk-docker-account-usage). But agreed your code will not push repos if they don't already exist, and the error message looks clear to me 👍

**Review by ChrisBAshton — APPROVED** [review]

**Pull request merged**